### PR TITLE
Unittest structure similarity

### PR DESCRIPTION
**Pull request by CunliangGeng**
- update and fix bugs in StuctureSimilarity.py
- add thorough unittests for StructureSimilarity.py

Note:  This PR contains #11, #13.

### CODE REVIEW

File: pdb2sql/StructureSimilarity.py [hunk 1]
```diff
@@ -252,34 +238,27 @@ def compute_irmsd_fast(self,izone=None,method='svd',cutoff=10,check=True):
         # read the izone file
         if izone is None:
             resData = self.compute_izone(cutoff,save_file=False)
-        elif not os.path.isfile(izone):
-            resData = self.compute_izone(cutoff,save_file=True,filename=izone)
+        # elif not os.path.isfile(izone):
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: why comment that ?

File: pdb2sql/StructureSimilarity.py [hunk 2]
```diff
@@ -96,48 +100,29 @@ def compute_lrmsd_fast(self,lzone=None,method='svd',check=True):
...
-                    ind_ref = atom_ref_short.index(at)
-                    xyz_decoy_short.append(data_decoy_short[ind_decoy][3:])
-                    xyz_ref_short.append(data_ref_short[ind_ref][3:])
-                except ValueError:
-                    warnings.warn(
-                        f'Decoy atom {at} not found in reference pdb')
+            data_decoy_long, data_decoy_short  = self.get_data_zone_backbone(self.decoy,resData,return_not_in_zone=True)
+            data_ref_long,   data_ref_short    = self.get_data_zone_backbone(self.ref,resData,return_not_in_zone=True)
+
+            atom_long = data_ref_long.intersection(data_decoy_long)
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: this intersection stuff does all the work ?

File: pdb2sql/StructureSimilarity.py [hunk 3]
```diff
@@ -299,9 +278,7 @@ def compute_irmsd_fast(self,izone=None,method='svd',cutoff=10,check=True):
         # return the RMSD
         return self.get_rmsd(xyz_contact_decoy,xyz_contact_ref)
 
-
-
-    def compute_izone(self,cutoff=5.0,save_file=True,filename=None):
+    def compute_izone(self,cutoff=10.0,save_file=True,filename=None):
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: why a cutoff of 10 now ?

File: pdb2sql/StructureSimilarity.py [hunk 4]
```diff
@@ -498,18 +472,19 @@ def compute_lrmsd_pdb2sql(self,exportpath=None,method='svd'):
...
+        backbone = ['CA', 'C', 'N', 'O']
+
         # create the sql
         sql_decoy = pdb2sql(self.decoy,sqlfile='decoy.db')
         sql_ref = pdb2sql(self.ref,sqlfile='ref.db')
 
         # extract the pos of chains A
-        xyz_decoy_A = np.array(sql_decoy.get('x,y,z',chainID='A'))
-        xyz_ref_A = np.array(sql_ref.get('x,y,z',chainID='A'))
+        xyz_decoy_A = np.array(sql_decoy.get('x,y,z',chainID='A', name=backbone))
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: why only backbone ?

File: pdb2sql/StructureSimilarity.py (hunk 1 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: `izone` is used as both input file and output file, which is not unique. I left a TODO comment around that to add output file to the function.

File: pdb2sql/StructureSimilarity.py (hunk 2 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: It gets the common residues or atoms and ignore missing ones. It's  actually checking if the sequences are consistent or not, but it cannot fix it if not consistent(even the former code cannot do this).

File: pdb2sql/StructureSimilarity.py (hunk 3 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: It should be 10Å for iRMSD and 5Å for FNAT. 
It's a bug but fine, the code invoking it resets the cutoff to 10.

File: pdb2sql/StructureSimilarity.py (hunk 4 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: Only backbone is needed in L-RMSD calculation, for both fitting and rmsd calculation. Here is the original reference https://doi.org/10.1002/prot.10393.

**Pull request merged**